File: src/Restaurant/FTGO.Restaurant.Tests/Generators.fs
namespace FTGO.Restaurant.Tests

open FsCheck
open FTGO.Common.BaseTypes

type Generators =

    static member NonEmptyString () =
        let map (FsCheck.NonEmptyString value) = value |> NonEmptyString.create |> Option.get
        { new Arbitrary<_> () with
            override _.Generator = Arb.generate |> Gen.map map
            override _.Shrinker value = value.Value |> FsCheck.NonEmptyString |> Arb.shrink |> Seq.map map }

File: src/AD/AD.InMemoryStore.Tests/Generators.cs
using FsCheck;
using System;
using System.Collections.Generic;
using System.Linq;

namespace AD.InMemoryStore.Tests
{
    static class Generators
    {
        class ValueTupleArbitrary<T1, T2> : Arbitrary<(T1, T2)>
        {
            static (T1, T2) ToValueTuple(Tuple<T1, T2> tuple) => (tuple.Item1, tuple.Item2);

            public override Gen<(T1, T2)> Generator =>
                Arb.Generate<Tuple<T1, T2>>().Select(ToValueTuple);

            public override IEnumerable<(T1, T2)> Shrinker((T1, T2) _arg1) =>
                Arb.Shrink(Tuple.Create(_arg1.Item1, _arg1.Item2)).Select(ToValueTuple);
        }

        public static Arbitrary<(T1, T2)> ValueTuple2<T1, T2>() => new ValueTupleArbitrary<T1, T2>();
    }
}
